thrifty-integration-tests/src/test/kotlin/com/microsoft/thrifty/integration/conformance/server/TestServer.kt

```diff
@@ -0,0 +1,131 @@
+/*
+ * Thrifty
+ *
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING
+ * WITHOUT LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE,
+ * FITNESS FOR A PARTICULAR PURPOSE, MERCHANTABLITY OR NON-INFRINGEMENT.
+ *
+ * See the Apache Version 2.0 License for specific language governing permissions and limitations under the License.
+ */
+package com.microsoft.thrifty.integration.conformance.server
+
+import com.microsoft.thrifty.integration.kgen.coro.ThriftTestProcessor
+import com.microsoft.thrifty.protocol.BinaryProtocol
+import com.microsoft.thrifty.protocol.CompactProtocol
+import com.microsoft.thrifty.protocol.JsonProtocol
+import com.microsoft.thrifty.protocol.Protocol
+import com.microsoft.thrifty.testing.ServerProtocol
+import com.microsoft.thrifty.testing.ServerTransport
+import com.microsoft.thrifty.transport.BufferTransport
+import com.microsoft.thrifty.transport.SocketTransport
+import com.microsoft.thrifty.transport.Transport
+import com.sun.net.httpserver.HttpContext
+import com.sun.net.httpserver.HttpServer
+import org.junit.jupiter.api.extension.AfterEachCallback
+import org.junit.jupiter.api.extension.BeforeEachCallback
+import org.junit.jupiter.api.extension.Extension
+import org.junit.jupiter.api.extension.ExtensionContext
+import java.net.InetAddress
+import java.net.InetSocketAddress
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.Executors
+
+import java.util.concurrent.ThreadPoolExecutor
+import java.io.OutputStream
+
+import java.io.IOException
+
+import com.sun.net.httpserver.HttpExchange
+import kotlinx.coroutines.runBlocking
+import okio.Buffer
+
+
+class TestServer @JvmOverloads constructor(private val protocol: ServerProtocol = ServerProtocol.BINARY) : Extension, BeforeEachCallback, AfterEachCallback {
```

Comment:
Curious at the use of `@JvmOverloads`.  Do you expect this to be used from Java?

Reply:
This file was based on the existing `TestServer.java` that I converted to Kotlin using IntelliJ, so it was so helpful as to automatically add the annotation :)
I've removed it now.